docs/en/guides/docker-quickstart.md

```diff
@@ -103,6 +109,62 @@ sudo docker run -it --ipc=host --gpus '"device=2,3"' $t
 
 The `-it` flag assigns a pseudo-TTY and keeps stdin open, allowing you to interact with the container. The `--ipc=host` flag enables sharing of host's IPC namespace, essential for sharing memory between processes. The `--gpus` flag allows the container to access the host's GPUs.
 
+### Visualize Ultralytics' Object detection in your GNU-Linux Display Server
+
+???+ warning "Highly Experimental"
+   The following instructions are highly experimental. Sharing a X11 socket with a docker container can raise security concerns, and as such this solution should be tested only in a controlled environment. 
+   More information on the subject [here](http://users.stat.umn.edu/~geyer/secure.html)
+
+Firstly, there is the need to allow the docker container to access the X11 socket of your GNU-Linux Display Server. There is also the need to mount the socket in the container and have the `DISPLAY` environment variable set to point to it.
+
+Secondly there is the need to grant access to the X11 Server. For that there is the need to allow connections from the Docker container into the host. A restrictive way to do it is :
+```bash
+xhost +local:docker
+```
+This allows connections from the `docker` group into the X11 Server. 
+
+You can later restore the access control with: 
+```bash
+xhost -local:docker
+```
+
+A simple way to validate that the docker group has access to the X11 server is to run a container with a GUI program like `xclock` or `xeyes`. Alternatively, you can also install these programs in the ultralytics docker container to test the access to the X11 Server of your GNU-Linux Display Server .
+
+Below, two examples are presented on how to run the Ultralytics Object detection container in both Xorg and Wayland using the interact mode `-it`. You can run the commands to start the docker container using a [GPU](#using-gpus)
+#### Xorg
+1. Open a terminal and run:
+
+```bash
+#Note 1: To debug the QT Plugins you can add the following flag to the command: -e QT_DEBUG_PLUGINS=1 
+#Note 2: Wayland also supports running the Xorg command.
+docker run -e DISPLAY=$DISPLAY -v /tmp/.X11-unix:/tmp/.X11-unix -v ~/.Xauthority:/root/.Xauthority -it $t
+```
+2. Open another terminal and run:
+```bash
+xhost +local:docker
+```
+3. Test the connection to the X11 Server inside the docker terminal:
+```bash
+yolo predict show=True
+```
+#### Wayland
+1. Open a terminal and use the following command to run the container:
+```bash
+# To debug the QT Plugins you can add the following flag to the command: -e QT_DEBUG_PLUGINS=1 
+docker run -e DISPLAY=$DISPLAY -v $XDG_RUNTIME_DIR/$WAYLAND_DISPLAY:/tmp/$WAYLAND_DISPLAY --net=host -it $t  
+
+```
+
+2. Open another terminal and run:
+```bash
+xhost +local:docker
+#Note: You may need to rerun the command each time that the X11 Server session closes. 
+```
+3. Test the connection to the X11 Server inside the docker terminal:
+```bash
+yolo predict show=True
+```
+
```

Comment:
```suggestion

        ## Running Ultralytics in Docker Container
        
        Here's how to execute the Ultralytics Docker container:
        ### Using only the CPU
        ```bash
        # Run with all GPUs
        sudo docker run -it --ipc=host $t
        ```
        ### Using GPUs
        ```bash
        # Run with all GPUs
        sudo docker run -it --ipc=host --gpus all $t
        
        # Run specifying which GPUs to use
        sudo docker run -it --ipc=host --gpus '"device=2,3"' $t
        ```
        
        The `-it` flag assigns a pseudo-TTY and keeps stdin open, allowing you to interact with the container. The `--ipc=host` flag enables sharing of host's IPC namespace, essential for sharing memory between processes. The `--gpus` flag allows the container to access the host's GPUs.
        
        ### Run graphical user interface (GUI) applications in a Docker Container
        
        ???+ danger "Caution: Highly Experimental"
        
            The following instructions are experimental. Sharing a X11 socket with a docker container poses potential security risks. Therefore, it's recommended to test this solution only in a controlled environment. For more information, refer to this [resource](http://users.stat.umn.edu/~geyer/secure.html).
        
        Docker is primarily used to containerize background applications and CLI programs, but it can also run graphical programs. In the Linux world, two main graphic servers handle graphical display: X11 and Wayland. Before starting it's essential to determine which graphics server you're currently using. Just run this command to find out:
        
        ```bash
        env | grep -E -i 'x11|xorg|wayland'
        ```
        
        !!! example ""
        
            === "X11"
        
                If you're using X11, you can run the following command to allow the Docker container to access the X11 socket:
        
                ```bash
                xhost +local:docker && docker run -e DISPLAY=$DISPLAY -v /tmp/.X11-unix:/tmp/.X11-unix -v ~/.Xauthority:/root/.Xauthority -it --ipc=host $t
        
                ```
        
                This command sets the `DISPLAY` environment variable to the host's display, mounts the X11 socket, and maps the `.Xauthority` file to the container. The `xhost +local:docker` command allows the Docker container to access the X11 server.
        
                ??? info "Use GPUs"
                    If you're using [GPUs](#using-gpus), you can add the `--gpus all` flag to the command.
        
        
            === "Wayland"
        
                For Wayland, use the following command:
        
                ```bash
                xhost +local:docker && docker run -e DISPLAY=$DISPLAY -v $XDG_RUNTIME_DIR/$WAYLAND_DISPLAY:/tmp/$WAYLAND_DISPLAY --net=host -it --ipc=host $t
                ```
        
                This command sets the `DISPLAY` environment variable to the host's display, mounts the Wayland socket, and allows the Docker container to access the Wayland server.
        
                ??? info "Use GPUs"
                    If you're using [GPUs](#using-gpus), you can add the `--gpus all` flag to the command.
        
        Now you can display graphical applications inside your docker container. For example, you can run the following command to visualize the predictions of the YOLO model:
        
        ```bash
        yolo predict show=True
        ```
        
        ???+ warning "Revoke access"
        
            In both cases, don't forget to revoke access from the docker group when you're done.
            ```bash
            xhost -local:docker
            ```
        
        ??? info "Testing"
        
            A simple way to validate that the docker group has access to the X11 server is to run a container with a GUI program like `xclock` or `xeyes`. Alternatively, you can also install these programs in the ultralytics docker container to test the access to the X11 server of your GNU-Linux Display Server. If you run into any problems, consider setting the environment variable `-e QT_DEBUG_PLUGINS=1`. This action enables the output of debugging information, aiding in the troubleshooting process.
```

Reply:
Yes I have validated the commands , in my personal setup sway (wayland) setup and another machine running ubuntu (Xorg).

Reply:
@ambitious-octopus  these are the specs of the machine: 
<details>
![image](https://github.com/ultralytics/ultralytics/assets/11836470/c7c5048a-286d-44f2-99f7-8cb801e525d8)
</details>

Reply:
Your suggestion looks good. I will wait for  @Burhan-Q input before I commit the suggestion.